File: src/main/java/org/exampleapps/greatbig/repository/TagRepository.java
package org.exampleapps.greatbig.repository;

import org.exampleapps.greatbig.domain.Tag;
import org.springframework.stereotype.Repository;

import org.springframework.data.jpa.repository.*;


/**
 * Spring Data JPA repository for the Tag entity.
 */
@SuppressWarnings("unused")
@Repository
public interface TagRepository extends JpaRepository<Tag,Long> {
    
}

File: src/main/kotlin/org/exampleapps/greatbig/repository/TagRepository.kt
package org.exampleapps.greatbig.repository

import org.exampleapps.greatbig.domain.Tag
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface TagRepository : CrudRepository<Tag, Long> {
    fun findByName(name: String): Tag?
}